lib/Modules/ComputeHooks/ComputeHooks.psm1

```diff
@@ -106,1360 +56,800 @@ function Install-Prerequisites {
+                    }
+                }
+            }
+            return
+        }
     }
 
-    $bond = New-NetLbfoTeam -Name $name -TeamMembers $bondPorts.Name -TeamNicName $name -TeamingMode LACP -Confirm:$false
-    $isUp = WaitFor-BondUp -bond $bond.Name
-    if (!$isUp){
-        Throw "Failed to bring up $name"
+    $oldSwitchName = Get-CharmState @stateParams
+    if($oldSwitchName) {
+        Remove-VMSwitch -Name $oldSwitchName -Force -Confirm:$false
+        Remove-CharmState @stateParams
     }
 
-    $adapter = Get-NetAdapter -Name $name
-    if(!$adapter){
-        Throw "Failed to find $name"
+    Write-JujuInfo "Adding new vmswitch: $vmSwitchName"
+
+    New-VMSwitch -Name $vmSwitchName -NetAdapterName $dataPort.Name -AllowManagementOS $managementOS | Out-Null
+    Set-CharmState @stateParams -Value $vmSwitchName
+}
+
+function Install-NovaFromZip {
+    Param(
+        [Parameter(Mandatory=$true)]
+        [string]$InstallerPath
+    )
+
+    if ((Test-Path $NOVA_INSTALL_DIR)) {
+        Remove-Item -Recurse -Force $NOVA_INSTALL_DIR
     }
-    $returnCode = Invoke-DHCPRenew $adapter
-    if($returnCode -eq 1) {
-        Invoke-JujuReboot -Now
+
+    Write-JujuInfo "Unzipping '$InstallerPath' to '$NOVA_INSTALL_DIR'"
+
+    Expand-ZipArchive -ZipFile $InstallerPath -Destination $NOVA_INSTALL_DIR | Out-Null
+
+    $configDir = Join-Path $NOVA_INSTALL_DIR "etc"
+    if (!(Test-Path $configDir)) {
+        New-Item -ItemType Directory $configDir | Out-Null
+        $distro = Get-OpenstackVersion
+        $templatesDir = Join-Path (Get-JujuCharmDir) "templates"
+        $policyFile = Join-Path $templatesDir "$distro\policy.json"
+        Copy-Item $policyFile $configDir | Out-Null
     }
-    return $name
-}
 
-function Get-TemplatesDir {
-    $charmDir = Get-JujuCharmDir
-    $templates =  Join-Path $charmDir "templates"
-    return $templates
+    New-ExeServiceWrapper | Out-Null
 }
 
-function Get-PackageDir {
-    $charmDir = Get-JujuCharmDir
-    $packages =  Join-Path $charmDir "packages"
-    return $packages
+function Install-NovaFromMSI {
+    Param(
+        [Parameter(Mandatory=$true)]
+        [string]$InstallerPath
+    )
+
+    $logFile = Join-Path $env:APPDATA "nova-installer-log.txt"
+    $extraParams = @("SKIPNOVACONF=1", "INSTALLDIR=`"$NOVA_INSTALL_DIR`"")
+    Install-Msi -Installer $InstallerPath -LogFilePath $logFile -ExtraArgs $extraParams
+
+    # Delete the Windows services created by default by the MSI,
+    # so the charm can create them later on.
+    $charmServices = Get-CharmServices
+    $serviceNames = $charmServices.Values | ForEach-Object { $_['service'] }
+    Remove-WindowsServices -Names $serviceNames
+
+    $product = Get-InstalledOpenStackProduct -Project "Nova"
+    if(!$product) {
+        Throw "Failed to get the installed OpenStack Nova product"
+    }
+    Set-CharmState -Namespace "novahyperv" -Key "product_name_installed" -Value $product.Name
```

Comment:
Sure. I could get the product using a regex. But I just wanted to use the exact name to get the product (no regex).